deploy/v2/ansible/roles/hana-system-replication/tasks/pre_checks.yml

```
@@ -0,0 +1,55 @@
+---
+
+# sapcontrol EXITCODES
+#     0  Last webmethod call successful
+#     1  Last webmethod call failed, invalid parameter
+#     2  StartWait, StopWait, WaitforStarted, WaitforStopped, RestartServiceWait
+#        timed out
+#        CheckSystemCertificates detected warnings
+#     3  GetProcessList succeeded, all processes running correctly
+#        CheckSystemCertificates detected errors
+#     4  GetProcessList succeeded, all processes stopped
+
+- name: Check HANA system replication variables are set and seem valid
+  assert:
+    that:
+      - "hana_group | trim | length != 0"
+      - "sid | trim | regex_search('^[A-Za-z][A-Za-z\\d][A-Za-z\\d]$') | length == 3"  # must be 3 characters <alpha><alnum><alnum>, e.g. HN1, not 1_H
+      - "instance_number | trim | regex_search('^\\d\\d$') | length == 2"  # must be 2 digits <num><num>, e.g. 00, not 0 or a0
+      - "hana_system_user | trim | length != 0"
+      - "hana_dir | trim | length != 0"
+      - "sap_dir | trim | length != 0"
+      - "tmp_ssfs_dir | trim | length != 0"
+
+- name: Check that HANA was installed successfully and is running normally
+  become_user: "{{ sid_adm_user }}"
+  shell: >
+    source ~/.bashrc
+    {{ sapcontrol_command }} -function GetProcessList
+  register: hana_process_list_result
+  failed_when: hana_process_list_result.rc != 3
+  changed: false
```

```suggestion
  changed_when: false
```
Otherwise you see error:
```
ERROR! 'changed' is not a valid attribute for a Task

The error appears to be in '/home/azureadm/sap-hana/deploy/v2/ansible/roles/hana-system-replication/tasks/pre_checks.yml': line 24, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:


- name: Check that HANA was installed successfully and is running normally
  ^ here

This error can be suppressed as a warning using the "invalid_task_attribute_failed" configuration
```

---

Outdated diff. I'll fix in the current branch. 👍 @rukawata